mango/clients/aemet.py

```diff
@@ -0,0 +1,598 @@
+import logging
+import os
+from datetime import datetime
+from math import radians, cos, sin, asin, sqrt
+from typing import Union, Any, Tuple
+
+from tqdm import tqdm
+
+from mango.clients.rest_client import RestClient
+from mango.shared import ApiKeyError
+from mango.validators.aemet import *
+
+
+class AemetClient(RestClient):
+    """
+    This class will handle the connection to the AEMET API. It will allow to get the meteorological data from the
+    meteorological stations in Spain. To initialize the class, an API key is needed. This key can be obtained in the
+    AEMET website: https://opendata.aemet.es/centrodedescargas/altaUsuario
+
+    :param api_key: API key to connect to the AEMET API
+    :param wait_time: Wait time between requests in seconds. Default is 1.25 seconds
+    :doc-author: baobab soluciones
+
+    Usage
+    -----
+
+    >>> import load_dotenv
+    >>> import os
+    >>> from mango.clients.aemet import AemetClient
+    >>> load_dotenv()
+    >>> client = AemetClient(os.environ["AEMET_API_KEY"])
+    """
+
+    def _set_up_default(self) -> None:
+        """
+        Do as function to allow for updates mid execution and set up default values
+        """
+        self._DEFAULT_WAIT_TIME = float(os.environ.get("AEMET_DEFAULT_WAIT_TIME", 1.25))
+        self._FETCH_STATIONS_URL = os.environ.get(
+            "AEMET_FETCH_STATIONS_URL",
+            "https://opendata.aemet.es/opendata/api/valores/climatologicos/inventarioestaciones/todasestaciones",
+        )
+        self._FETCH_MUNICIPIOS_URL = os.environ.get(
+            "AEMET_FETCH_MUNICIPIOS_URL",
+            "https://opendata.aemet.es/opendata/api/maestro/municipios",
+        )
+        self._FETCH_HISTORIC_URL = os.environ.get(
+            "AEMET_FETCH_HISTORIC_URL",
+            "https://opendata.aemet.es/opendata/api/valores/climatologicos/diarios/datos/fechaini"
+            "/{start_date_str}/fechafin/{end_date_str}"
+            "/estacion/{station}",
+        )
+        self._FETCH_DAILY_URL = os.environ.get(
+            "AEMET_FETCH_DAILY_URL",
+            "https://opendata.aemet.es/opendata/api/observacion/convencional/datos/estacion/{station}",
+        )
+        self._FETCH_FORECAST_URL = os.environ.get(
+            "AEMET_FETCH_FORECAST_URL",
+            "https://opendata.aemet.es/opendata/api/prediccion/especifica/municipio/diaria/{postal_code}",
+        )
+
+    def __init__(self, *, api_key: str, wait_time: Union[float, int] = None):
+        super().__init__()
+        self._set_up_default()
+        self._api_key = api_key
+        self._wait_time = wait_time or self._DEFAULT_WAIT_TIME
+        logging.getLogger("root")
```

Comment:
Is there any case we want to use another logger?

Reply:
How would you handle that? I took this from the Arcgis client

Reply:
I think it could be something like that:
```
if logger_name is None:
    self.logger = get_basic_logger()
else:
    self.logger = logging.getLogger(logger_name)
```
and add logger_name as an arg

Reply:
I do not llike to pass the logger name as a attribute to the method

Reply:
you could also have the class always created with root logger and add a method set_logger to replace the logger with another one if needed.